apis/python/src/tiledbsoma/dataframe.py

```diff
@@ -51,16 +53,19 @@ def create(
         self,
         schema: pa.Schema,
         index_column_names: Sequence[str] = (SOMA_JOINID,),
-        platform_config: Optional[PlatformConfig] = None,
+        platform_config: Optional[TileDBCreateOptions] = None,
```

Comment:
The `platform_config` parameter is designed to be extensible for future operations and (as of right now) we want to be able to accept the same `PlatformConfig` value across multiple calls (e.g. a user could pass the same `platform_config` dict to create and query etc.). Right now it’s only *used* on `create` but we’ll be adding more integration points (cf. the fact that the [format description](https://github.com/single-cell-data/TileDB-SOMA/tree/main/apis/python#platform_config-format) has only the `tiledb.create` field available.

*Internally* we can transform it into something more compact (which was the goal with the way I implemented the thing which wraps dictionaries, though it is admittedly somewhat inelegant).

Reply:
Since the somacore spec is just asking for a `Mapping`-typed param, would it be wrong to introduce a new `TileDbQueryOptions(soma.PlatformConfig)` class for query methods? Consider that both create() and query() methods (et. al.) would still adhere to the the somaspec (taking a soma.PlatformConfig-derived type), even though in this TileDB platform implementation, we'd be using separate classes for these individual cases.

Reply:
> a user could pass the same platform_config dict to create and query 
So there could _per-call_ TileDB options that are not create- or query-specific? I was imaging that any options that are not create- or query-specific would be specified as part of the SessionContext. But of course, that implies session-long settings, not call-specific setting.

Reply:
Makes sense re:allowing multiple platform impl options to be specified together. I'll re-introduce that.

What I'm not understanding is why we would allow both "create" and "read" sibling dicts into a `create()` call. The "read" sub-dict wouldn't be needed here, presumably. I was imagining platform_config would be specific to the operation being called.

Reply:
We went with something like this for a few reasons:

1. It would allow a library to use the same `platform_config` object across multiple calls if need be, particularly for complex operations that might have multiple steps (create an array, fill it up, etc.).
2. If two operations have a similarly-named setting that does something different, it won’t be confused between operations (something like read buffer size vs. write buffer size).

Reply:
@thetorpedodog Thanks for explaining those design justifications.  I've reinstated support for the original structure of platform_config. Have a look when you can, thanks!